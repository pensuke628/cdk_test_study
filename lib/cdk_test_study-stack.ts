import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as s3 from 'aws-cdk-lib/aws-s3';
// import { StaticSite } from './construct/static_site';

export class CdkTestStudyStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // new StaticSite(this, 'StaticSite', {
    //   staticContentPath: './app',
    // });

    const vpc = new ec2.Vpc(this, 'CdkTestStudyVpc', {
      maxAzs: 2,
      natGateways: 0,
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'CdkTestStudyPrivate',
          subnetType: ec2.SubnetType.PRIVATE_ISOLATED
        }
      ]
    });
    cdk.Tags.of(vpc).add('CreatedBy', 'CDK');

    const Bucket = new s3.Bucket(this, 'CdkTestStudyBucket1', {
      bucketName: 'hogehoge',
      enforceSSL: true,
      objectLockEnabled: true,
      versioned: true,
    });

    new s3.Bucket(this, 'CdkTestStudyBucket2', {
      objectLockEnabled: true,
      removalPolicy: cdk.RemovalPolicy.RETAIN
    });

    // S3のBucketNameをoutput
    new cdk.CfnOutput(this, 'S3BucketName', {
      value: Bucket.bucketName,
    });
  }
}
