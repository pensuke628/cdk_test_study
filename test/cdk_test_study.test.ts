import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { CdkTestStudyStack } from '../lib/cdk_test_study-stack';

// assertion test
describe('CdkTestStudyStack', () => {
  let template: Template;

  beforeAll(() => {
    const app = new cdk.App();
    const stack = new CdkTestStudyStack(app, "MyTestStack");
    // テンプレートの⽣成
    template = Template.fromStack(stack);
  });

  // snapshot
  test('snapshot', () => {
    expect(template.toJSON()).toMatchSnapshot();
  })

  // VPC
  test('VPCが1つ存在すること', () => {
    template.resourceCountIs("AWS::EC2::VPC", 1);
  });

  test('VPCのCIDRが10.0.0.0/16であること', () => {
    template.hasResourceProperties("AWS::EC2::VPC", 
      {
        CidrBlock: "10.0.0.0/16",
      }
    );
  });

  // Subnet
  // テストを2つ書く場合
  // test('Subnetが2つ存在すること', () => {
  //   template.resourceCountIs("AWS::EC2::Subnet", 2);
  // });

  // test('SubnetはすべてPrivate Subnetであること', () => {
  //   template.allResourcesProperties("AWS::EC2::Subnet",
  //     {
  //       MapPublicIpOnLaunch: false,
  //     }
  //   );
  // });

  // 1つのテストにまとめる場合
  test("Private Subnetが2つ存在する", () => {
    template.resourcePropertiesCountIs("AWS::EC2::Subnet", {
      MapPublicIpOnLaunch: false,
    }, 2);
  });

  // FlowLog
  // nagに対応する場合、VpcFlowLogを追加する
  // test ('VpcFlowLogが1つ存在すること', () => {
  //   template.resourcePropertiesCountIs("AWS::EC2::FlowLog", {
  //     ResourceType: "VPC",
  //   }, 1);
  // });
})
