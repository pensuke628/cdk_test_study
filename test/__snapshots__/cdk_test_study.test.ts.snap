// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdkTestStudyStack snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet1RouteTableAssociationAD59FD5B": {
      "Properties": {
        "RouteTableId": {
          "Ref": "CdkTestStudyVpcCdkTestStudyPrivateSubnet1RouteTableBB6508F0",
        },
        "SubnetId": {
          "Ref": "CdkTestStudyVpcCdkTestStudyPrivateSubnet1Subnet8D6AE612",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet1RouteTableBB6508F0": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "MyTestStack/CdkTestStudyVpc/CdkTestStudyPrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "CdkTestStudyVpcD7D81CBC",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet1Subnet8D6AE612": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "CdkTestStudyPrivate",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "MyTestStack/CdkTestStudyVpc/CdkTestStudyPrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "CdkTestStudyVpcD7D81CBC",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet2RouteTableAssociation9690DDA4": {
      "Properties": {
        "RouteTableId": {
          "Ref": "CdkTestStudyVpcCdkTestStudyPrivateSubnet2RouteTableEAA6E089",
        },
        "SubnetId": {
          "Ref": "CdkTestStudyVpcCdkTestStudyPrivateSubnet2SubnetEF7C3034",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet2RouteTableEAA6E089": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "MyTestStack/CdkTestStudyVpc/CdkTestStudyPrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "CdkTestStudyVpcD7D81CBC",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "CdkTestStudyVpcCdkTestStudyPrivateSubnet2SubnetEF7C3034": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.1.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "CdkTestStudyPrivate",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "MyTestStack/CdkTestStudyVpc/CdkTestStudyPrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "CdkTestStudyVpcD7D81CBC",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "CdkTestStudyVpcD7D81CBC": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "MyTestStack/CdkTestStudyVpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
